### PR TITLE
Create and use non-default service accounts in GKE

### PR DESCRIPTION
**Pull request by annuay-google**
## Approach
- Allow user to create new SA from blueprint and use the same in cluster and node pool
- Do not fetch default SA from a data source, instead construct using project id
- User can pass a pre existing SA with appropriate permissions. Checking for the right permissions is not in the scope of this PR and will be covered in a follow up PR

## Testing
- [x] Deployed hpc-gke.yaml with non default new SA and verified provisioning and destruction, and that SA with appropriate roles was created. Verified all node pools used the SA specified
- [x] Deployed hpc-gke.yaml with non default pre-existing SA and verified provisioning and destruction, and that no new SA was created. Verified all node pools used the SA specified
- [x] Deployed hpc-gke.yaml with no SA passed. Verified default SA was used in all node pools
- [x] Verified all GKE integration tests passed
- [x] Verified logs are pushed from k8s pods and visible in GCP logging from Pantheon

### Create and use new SA (one per node pool) - recommended
Check updated example blueprints
### Use pre-existing SA
```
  - id: gke_cluster
    source: modules/scheduler/gke-cluster
    use: [network1]
    settings:
      enable_private_endpoint: false  # Allows for access from authorized public IPs
      service_account_email: ml-01-gke-sa@ns-playground-a.iam.gserviceaccount.com
    outputs: [instructions]

  - id: compute_pool
    source: modules/compute/gke-node-pool
    use: [gke_cluster]
    settings:
      service_account_email: ml-01-gke-sa@ns-playground-a.iam.gserviceaccount.com
```
### Use default SA
```
  - id: gke_cluster
    source: modules/scheduler/gke-cluster
    use: [network1]
    settings:
      enable_private_endpoint: false  # Allows for access from authorized public IPs
    outputs: [instructions]

  - id: compute_pool
    source: modules/compute/gke-node-pool
    use: [gke_cluster]
```

### Submission Checklist

NOTE: Community submissions can take up to 2 weeks to be reviewed.

Please take the following actions before submitting this pull request.

* Fork your PR branch from the Toolkit "develop" branch (not main)
* Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development)
* Confirm that "make tests" passes all tests
* Add or modify unit tests to cover code changes
* Ensure that unit test coverage remains above 80%
* Update all applicable documentation
* Follow Cluster Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)

### CODE REVIEW

File: examples/ml-gke.yaml [hunk 1]
```diff
@@ -40,19 +40,32 @@ deployment_groups:
         - range_name: services
           ip_cidr_range: 10.0.32.0/20
 
+  - id: gke_service_account
+    source: community/modules/project/service-account
+    settings:
+      name: gke-service-account
+      project_roles:
+      - logging.logWriter
```
Comment (ankitkinra) [inline examples/ml-gke.yaml:48]: Would we not need more permissions like compute.createResources, networking permissions, etc?

Comment (annuay-google) [reply examples/ml-gke.yaml:48]: AFAIK, that's not required. We're passing the service to the GKE cluster (and GKE node pool), which uses this SA in the node_config argument. This is the SA the nodes will use. [parameter definition](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster#service_account). I'm running all GKE integration tests to see if there's something I've not caught

File: modules/compute/gke-node-pool/main.tf [hunk 2]
```diff
@@ -20,8 +20,6 @@ locals {
 }
 
 locals {
-  sa_email = var.service_account_email != null ? var.service_account_email : data.google_compute_default_service_account.default_sa.email
```
Comment (sharabiani) [inline modules/compute/gke-node-pool/main.tf:23]: Are we going to still use the default_service_account if the gke_service_account not provided?
If yes I believe we should keep the current logic and resources.

File: examples/storage-gke.yaml [hunk 3]
```diff
@@ -38,9 +38,33 @@ deployment_groups:
         - range_name: services
           ip_cidr_range: 10.0.32.0/20
 
+  - id: gke_service_account
```
Comment (sharabiani) [inline examples/storage-gke.yaml:41]: Isn't `gcp_service_account` more descriptive? GKE service accounts are to manage k8s resources while the account we need is GCP IAM sa.

File: examples/ml-gke.yaml (hunk 1 above)
Comment (ankitkinra) [reply examples/ml-gke.yaml:48]: Ack, thanks!

**Review by ankitkinra — APPROVED** [review]

File: modules/compute/gke-node-pool/main.tf (hunk 2 above)
Comment (sharabiani) [reply modules/compute/gke-node-pool/main.tf:23]: please ignore

File: examples/storage-gke.yaml (hunk 3 above)
Comment (sharabiani) [reply examples/storage-gke.yaml:41]: please ignore

**Pull request merged**